common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts

```diff
@@ -0,0 +1,72 @@
+import path = require("path");
+import * as tl from 'azure-pipelines-task-lib/task';
+import { getSystemAccessToken } from "./webapi";
+import fetch from "node-fetch";
+
+tl.setResourcePath(path.join(__dirname, 'module.json'), true);
+
+export async function getFederatedWorkloadIdentityCredentials(serviceConnectionName: string, tenantId?: string) : Promise<string>
+{
+    try {
+        let tenant = tenantId ?? tl.getEndpointAuthorizationParameterRequired(serviceConnectionName, "TenantId");
+        tl.debug(tl.loc('Info_UsingTenantId', tenantId));
+        const systemAccessToken = getSystemAccessToken();
+        const url = process.env["SYSTEM_OIDCREQUESTURI"]+"?api-version=7.1&serviceConnectionId="+serviceConnectionName;
+        
+        const ADOResponse: {oidcToken: string} = await (await fetch(url, 
+        {
+            method: 'POST', 
+            headers: 
+            {
+                'Content-Type': 'application/json',
+                'Authorization': 'Bearer '+ systemAccessToken
+            }
+        })).json() as {oidcToken: string};
+
+        tl.setSecret(ADOResponse.oidcToken);
+        let entraURI = "https://login.windows-ppe.net/"+tenant+"/oauth2/v2.0/token";
+        let clientId = tl.getEndpointAuthorizationParameterRequired(serviceConnectionName, "ServicePrincipalId");
+
+        let body = {
+            'scope': "499b84ac-1321-427f-aa17-267ca6975798/.default",
+            'client_id': clientId,
+            'client_assertion_type': "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+            'client_assertion': ADOResponse.oidcToken,
+            'grant_type': "client_credentials"
+        };
```

Comment:
nit: do you feel it's worth moving the hard-coded values in this body to a constants file? I know something like the `scope` you've provided is the resource for Azure DevOps when acquiring a token, so that might be something worth explicitly specifying in a constants file and sharing across other auth common code here? (similar comment may apply to the other hard-coded values, I don't know how expansive their scenarios are)

Reply:
I really dont expect these to be used anywhere else but they are somewhat magic strings